vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/LinkManager.java

```diff
@@ -36,37 +38,82 @@ public abstract class LinkManager<L extends ILink<T>, T extends IVSpaceElement,
 
     @Autowired
     private IStorageEngine storage;
+    
+    @Autowired
+    private IImageService imageService;
+
+    protected abstract void deleteLinkRepo(L link);
+
+    protected abstract void deleteLinkDisplayRepo(L link);
+
+    protected abstract void removeFromLinkList(ISpace space, L link);
+
+    protected abstract U updateLinkAndDisplay(L link, U displayLink);
+
+    protected abstract U getDisplayLink(String linkDisplayId) throws LinkDoesNotExistsException;
+
+    protected abstract L getLink(String linkId);
+
+    protected abstract L createLinkObject(String title, String id) throws SpaceDoesNotExistException;
 
+    protected abstract T getTarget(String linkedId);
+
+    protected abstract U createDisplayLink(L link);
+    
     @Override
     public U createLink(String title, String id, float positionX, float positionY, int rotation, String linkedId,
-            String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename)
-            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+            String linkLabel, String linkDesc, DisplayType displayType, byte[] linkImage, String imageFilename, String existingImageId)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, ImageDoesNotExistException {
 
+        if (title == null || title.trim().isEmpty()) {
+            throw new IllegalArgumentException("Title cannot be null or empty.");
```

Comment:
this is not the right exception to throw; it's a runtime exception so it does not need to be caught and handled, because it would be thrown if it's an argument that simply can't be dealt with. Here, it can definitely be dealt with if the title is null (simply ask the user to enter a title). Also, I don't see why the title couldn't be empty. If the user, for instance, adds an image, there does not need to be title to make this work.